omod-1.9/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/controller/openmrs1_9/FrontendJsonConfigController1_9.java

```diff
@@ -0,0 +1,128 @@
+    private static final Logger log = LoggerFactory.getLogger(FrontendJsonConfigController1_9.class);
+
+    @RequestMapping(method = RequestMethod.GET)
+	public void getFrontendConfigFile(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        File jsonConfigFile = getJsonConfigFile();
+        if (!jsonConfigFile.exists()) {
+            log.debug("Configuration file does not exist");
+            response.sendError(HttpServletResponse.SC_NOT_FOUND, "Configuration file does not exist");
+        }
+        try {
+            InputStream inputStream = new FileInputStream(jsonConfigFile);
+            OutputStream outStream = response.getOutputStream();
+            OpenmrsUtil.copyFile(inputStream, outStream);
+    
+            response.setContentType("application/json");
+            response.setHeader("Content-Disposition", "attachment; filename=" + jsonConfigFile.getName());
+            response.setStatus(HttpServletResponse.SC_OK);
+        } catch (IOException e) {
+            log.error("Error reading Configuration file: " + jsonConfigFile.getAbsolutePath(), e);
+            response.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR, "Error reading Configuration file: " + jsonConfigFile.getPath());
+        }
+	}
+
+    @RequestMapping(method = RequestMethod.POST)
+    @ResponseStatus(HttpStatus.OK)
+    public void saveFrontendConfigFile(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        User user = Context.getAuthenticatedUser();
+        if (user == null || !user.isSuperUser()) {
+            log.error("Authorization error while creating a config.json file");
+            response.sendError(HttpServletResponse.SC_FORBIDDEN, "Authorization error. Admin privileges required to save the config.json file");
+            return;
+        }
+        saveJsonConfigFile(request, response);
+    }
+
+    private void saveJsonConfigFile(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        File jsonConfigFile = getJsonConfigFile();
+        InputStream inputStream = request.getInputStream();
+        String requestBody = IOUtils.toString( inputStream , "UTF-8");
+        try {
+            // verify that is in a valid JSON format
+            new ObjectMapper().readTree(requestBody);
+        } catch (JsonProcessingException e) {
+            log.error("Invalid JSON format", e);
+            response.sendError(HttpServletResponse.SC_BAD_REQUEST, "Invalid JSON format");
+        }
+        inputStream = new ByteArrayInputStream(requestBody.getBytes(StandardCharsets.UTF_8));
+        OutputStream outStream = Files.newOutputStream(jsonConfigFile.toPath());
+        OpenmrsUtil.copyFile(inputStream, outStream);
+
+        if (jsonConfigFile.exists()) {
+            log.debug("file: '{}' written successfully", jsonConfigFile.getAbsolutePath());
+            response.setStatus(HttpServletResponse.SC_OK);
+        }
```

Comment:
Never end a function like this with an `if` without an `else`. What response would the consumer get if the `jsonConfigFile.exists()` check returned `false`? (This is just intended as a pedagogic remark; as I said above, the entire `if` is unnecessary